pkg/topology/run.go

```diff
@@ -537,6 +540,50 @@ func (tj *TopologyJob) Run(job job.JobRuntime) error {
 
 	query.FlushTopologyCache()
 
+	if t.Spec.PushLocation.URL != "" {
+		for _, c := range results {
+			if err := pushTopologyToLocation(job.Context, t.Spec.PushLocation, *c); err != nil {
+				job.History.AddErrorf("error pushing topology: %v", err)
+			}
+		}
+	}
+
+	return nil
+}
+
+func pushTopologyToLocation(ctx dutyCtx.Context, target v1.PushLocation, comp pkg.Component) error {
+	ep := fmt.Sprintf("%s/push/topology", target.URL)
+	client := http.NewClient()
+	if !target.Auth.Username.IsEmpty() {
+		user, err := ctx.GetEnvValueFromCache(target.Auth.Username, ctx.GetNamespace())
+		if err != nil {
+			return fmt.Errorf("error getting username for topology[%s] push: %w", comp, err)
+		}
+
+		password, err := ctx.GetEnvValueFromCache(target.Auth.Password, ctx.GetNamespace())
+		if err != nil {
+			return fmt.Errorf("error getting password for topology[%s] push: %w", comp, err)
+		}
+
+		client = client.Auth(user, password)
+	}
+
+	req := client.R(ctx).
+		Header(echo.HeaderContentType, echo.MIMEApplicationJSON)
+
+	if canary.UpstreamConf.Valid() {
```

Comment:
Not sure that these should be linked - we can add the agent name to the url in the chart